backend/api.js

```diff
@@ -19,15 +11,37 @@ const foundParams = (params, wantedParam) => {
     return false;
 };
 
-export function v1(requestEvent, sequelizeManager, callback) {
+const obtainDatabaseForTask = (
+    requestEvent, sequelizeManager, sessionSelected, callback
+) => {
+    let database = null;
+    if (foundParams(requestEvent.params, 'database')) {
+        // check if there is a current database in use
+        database = requestEvent.params.database;
+    } else if (sequelizeManager.sessions[sessionSelected]) {
+        database = sequelizeManager.sessions[sessionSelected].config.database;
```

Comment:
we may want to be stricter in our API around this. if the user didn't supply a `database` in the call then we raise an error no matter what instead of defaulting to the current session. a stricter API will be easier to write different implementations around in the future (the writer won't need to handle these different conditions)